pycti/connector/opencti_connector_helper.py

```diff
@@ -308,28 +308,32 @@ def run(self) -> None:
                 self.pika_connection = pika.BlockingConnection(self.pika_parameters)
                 self.channel = self.pika_connection.channel()
                 try:
+                    # confirm_delivery is only for cluster mode rabbitMQ
+                    # when not in cluster mode this line raise an exception
                     self.channel.confirm_delivery()
                 except Exception as err:  # pylint: disable=broad-except
-                    self.helper.connector_logger.warning(str(err))
+                    self.helper.connector_logger.debug(str(err))
                 self.channel.basic_qos(prefetch_count=1)
                 assert self.channel is not None
                 self.channel.basic_consume(
                     queue=self.queue_name, on_message_callback=self._process_message
                 )
                 self.channel.start_consuming()
-            except (KeyboardInterrupt, SystemExit):
-                self.channel.stop_consuming()
-                self.pika_connection.close()
-                self.helper.connector_logger.info("Connector stop")
-                sys.exit(0)
             except Exception as err:  # pylint: disable=broad-except
-                self.pika_connection.close()
+                try:
+                    self.pika_connection.close()
+                except Exception as errInException:
+                    self.helper.connector_logger.debug(
+                        type(errInException).__name__, {"reason": str(errInException)}
+                    )
                 self.helper.connector_logger.error(
                     type(err).__name__, {"reason": str(err)}
                 )
-                sys.exit(1)
```

Comment:
this was the main issue: the thread was stopped, but not the whole python process

Reply:
So catching this we keep the loop alive and connector continue to try to reconnect. Fine for me.
Did you manage to validate that the thread is calling the stop function when system ask for interruption?

Reply:
FYI `docker compose stop` work as before